aips/resource_groups.md

```diff
@@ -75,31 +73,33 @@ The motivation of using a `struct` is that
 2. It can build upon the existing storage model that knows how to read and write data stored at  `StructTag`s. Thus it limits the implementation impact to the VM and readers of storage, storage can remain agnostic to this change.
 3. Only `struct` and `fun` can have attributes, which in turn let’s us define additional parameters like `scope`.
 
-Each entry in a resource group is identified by the `resource_group` attribute:
+Each entry in a resource group is identified by the `resource_group_member` attribute:
 
 ```move
-#[resource_group(container = aptos_framework::object::ObjectGroup)]
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
 struct Object has key {
     guid_creation_num: u64,
 }
 
-#[resource_group(container = aptos_framework::object::ObjectGroup)]
+#[resource_group_member(group = aptos_framework::object::ObjectGroup)]
 struct Token has key {
     name: String,
 }
 ```
 
 During compilation and publishing, these attributes are checked to ensure that:
 
-1. A `resource_group_container` has no abilities and no fields.
-2. The `scope` within the `resource_group_container` can only become more permissive, that is it can either remain at a remain at the same level of accessibility or increase to the next.
-3. Each entry within a resource group has a `resource_group` attribute.
-4. The `container` parameter is set to a struct that is labeled as a `resource_group_container`.
-5. During upgrade, an existing `struct` cannot either enter or leave a `resource_group`.
+1. A `resource_group_member` has no abilities and no fields.
+2. The `scope` within the `resource_group_member` can only become more permissive, that is it can either remain at a remain at the same level of accessibility or increase to the next.
+3. Each entry within a resource group has a `resource_group_member` attribute.
+4. The `group` parameter is set to a struct that is labeled as a `resource_group`.
+5. During upgrade, an existing `struct` cannot either enter or leave a `resource_group_member`.
```

Comment:
I read the AIP but I'm still not sure why we have these constraints? I figure it's so we don't have to deal with backfilling / null data respectively?

Reply:
you'll end up in a situation where data is lost, I'll make that clearer. Specifically, if a struct enters a resource group any existing data will become inaccessible. Now one could theoretically add key to something after the fact and potentially make that into a resource group. But key + resource group go hand in hand.